### ISSUE TITLE
Implicit conversions are defined but warn about assignments to the same variable

### ISSUE THREAD

**Issue by ilyfairy** [dotnet/roslyn#78170]
**Version Used**: 10.0.0-preview.3.25171.5

**Steps to Reproduce**:

https://sharplab.io/#v2:C4LgTgrgdgPg8lApgHgPYCMBWiDGwB8ABKkoQLyFIDuxWuwAFAJQDcAsAFAmLmEMbY8Tbuw6cAAgGZC4gEyEEKACr5OAb06EtM6Yubki4gIwA6YwE4GAImpXWm7VJlGAbIQCWAWwAOAG3c47sDE3ohgAIbAqGAKSMgqDEqEAG7hvhCITAaUiDSK8fjMogC+QA===

```cs
One<object> one = new object();
one = (object)one;

public class One<T>
{
    public One() => System.Console.WriteLine("new");
    public static implicit operator One<T>(T value) => new One<T>();
}
```

**Diagnostic Id**: CS1717

**Expected Behavior**:No warning shown

**Actual Behavior**:CS1717 warning is shown
